octoprint_TimeToFilament/__init__.py

```diff
@@ -66,6 +79,33 @@ def on_startup(self, host, port):
   @octoprint.plugin.BlueprintPlugin.route("/get_settings_defaults", methods=["GET"])
   def get_settings_defaults_as_string(self):
     return json.dumps(self.get_settings_defaults())
+    
+  def is_blueprint_csrf_protected(self):
+    return True
+    
+  @octoprint.plugin.BlueprintPlugin.route("/", methods=["GET"])
+  def get_cache(self):
+    if self._printer.get_state_id() not in (["PRINTING", "PAUSED"]):
+      return flask.jsonify(None)
+    addData = self.additional_state_data(False)
+    dispLines = self._settings.get(["displayLines"])
+    for idx, line in enumerate(dispLines):
+      dispLines[idx] = dict_merge(line, addData[line["regex"]])
+      if "timeLeft" in dispLines[idx]:
+        dispLines[idx]["timeLeft"] = max(self._printer.get_current_data()["progress"]["printTimeLeft"]-dispLines[idx]["timeLeft"],0)
+        import datetime
+        dispLines[idx]["timeLeftFormatted"] = str(datetime.timedelta(seconds=dispLines[idx]["timeLeft"])).split('.', 2)[0]
+        dispLines[idx]["dateTimeFinished"] = (datetime.datetime.now()-datetime.timedelta(seconds=dispLines[idx]["timeLeft"])).strftime("%H:%M:%S")
+      if "searchPos" in dispLines[idx]:
+        dispLines[idx]["searchPos"] = self._printer._comm._currentFile.getFilepos()
+      from jinja2 import Environment, BaseLoader
+      env = Environment(loader=BaseLoader()).from_string(dispLines[idx]["formatExternal"])
+      variables = dict_merge(dispLines[idx], __builtins__)
```

Comment:
Based on this usage, it's not clear to me why this is even a class variable...  Could it not just be a local variable?

Reply:
It should be! I will rename it to `jinja_variables` and use it as an local variable.